src/components/gcds-button/gcds-button.e2e.ts

```diff
@@ -18,26 +18,26 @@ describe('gcds-button', () => {
 
 describe('gcds-button a11y tests', () => {
 
-  it('Colour contrast: Primary task, variants and states', async () => {
+  it('Colour contrast: Primary button-role, button-styles and interaction-states', async () => {
     const page = await newE2EPage();
     await page.setContent(`
-      <gcds-button task="primary">Button Label</gcds-button>
-      <gcds-button task="primary" state="hover">Button Label</gcds-button>
-      <gcds-button task="primary" state="active">Button Label</gcds-button>
-      <gcds-button task="primary" state="focus">Button Label</gcds-button>
-      <gcds-button task="primary" state="disabled">Button Label</gcds-button>
-
-      <gcds-button task="primary" variant="outline">Button Label</gcds-button>
-      <gcds-button task="primary" variant="outline" state="hover">Button Label</gcds-button>
-      <gcds-button task="primary" variant="outline" state="active">Button Label</gcds-button>
-      <gcds-button task="primary" variant="outline" state="focus">Button Label</gcds-button>
-      <gcds-button task="primary" variant="outline" state="disabled">Button Label</gcds-button>
-
-      <gcds-button task="primary" variant="text-only">Button Label</gcds-button>
-      <gcds-button task="primary" variant="text-only" state="hover">Button Label</gcds-button>
-      <gcds-button task="primary" variant="text-only" state="active">Button Label</gcds-button>
-      <gcds-button task="primary" variant="text-only" state="focus">Button Label</gcds-button>
-      <gcds-button task="primary" variant="text-only" state="disabled">Button Label</gcds-button>
+      <gcds-button button-role="primary">Button Label</gcds-button>
+      <gcds-button button-role="primary" interaction-state="hover">Button Label</gcds-button>
```

Comment:
The interaction-state tests can be removed if we are removing the interaction-state.